roles/endpoints/defaults/main.yml

```diff
@@ -0,0 +1,164 @@
+      admin: https://{{ fqdn }}:8004                                                                       
+      internal: https://{{ fqdn }}:8004                                                                    
+      public: https://{{ fqdn }}:8004
+      path: v1/%(tenant_id)s                                                                      
+    port:                                                                          
+      backend_api: 8005                                                                 
+      haproxy_api: 8004
+  heat_cfn:
+    name: heat-cfn
+    type: cloudformation
+    description: 'Heat Cloudformation API'
+    version: v1
+    url:                                                                           
+      admin: https://{{ fqdn }}:8000                                                                       
+      internal: https://{{ fqdn }}:8000                                                                   
+      public: https://{{ fqdn }}:8000                                                                      
+    port:                                                                          
+      backend_api: 8001                                                                 
+      haproxy_api: 8000
+  ceilometer:
+    name: ceilometer
+    type: Telemetry
+    description: 'Telemetry Service'
+    url:                                                                           
+      admin: https://{{ fqdn }}:8777                                                                       
+      internal: https://{{ fqdn }}:8777                                                                    
+      public: https://{{ fqdn }}:8777                                                                      
+    port:                                                                          
+      backend_api: 9777                                                                 
+      haproxy_api: 8777
+  ironic:
+    name: ironic
+    type: baremetal
+    description: 'Ironic bare metal provisioning service'
+    url:                                                                           
+      admin: https://{{ fqdn }}:6385                                                                    
+      internal: https://{{ fqdn }}:6385                                                                     
+      public: https://{{ fqdn }}:6385                                                                       
+    port:                                                                          
+      backend_api: 6384                                                                 
+      haproxy_api: 6385
+  swift:
+    name: swift
+    type: object-storage
+    description: 'Object Storage Service'
+    version: v1
+    url:
+      admin: https://{{ fqdn }}:8090
+      internal: https://{{ undercloud_floating_ip }}:8090
```

Comment:
swift internal url should not use undercloud_floating_ip. HAProxy VIP parameter was created to be set to Swift ucarp ip